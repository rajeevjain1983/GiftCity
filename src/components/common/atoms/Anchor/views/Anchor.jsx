// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import {
  buildUrl,
  getAsPathWithSlug,
  getMappedPageHref,
  triggerPostMessage,
  getAPIConfig,
} from '../../../../../utils';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

const getAsLinkPath = (IsSlugPathAdded, asPath, incomingUrl) => {
  return IsSlugPathAdded ? asPath || incomingUrl : getAsPathWithSlug(asPath || incomingUrl);
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  asPath,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  url,
  text,
  dataLocator,
  IsSlugPathAdded,
  ...other
}) => {
  const targetVal = target || '_self';
  let incomingUrl = to || url;
  const asLinkPath = getAsLinkPath(IsSlugPathAdded, asPath, incomingUrl);
  if (!noLink) {
    incomingUrl = getMappedPageHref(incomingUrl);
  }
  const hrefUrl = asLinkPath || buildUrl(incomingUrl);

  const apiConfigObj = getAPIConfig();

  let AnchorComponent = null;
  if (children || text) {
    AnchorComponent = noLink ? (
      <a
        href={buildUrl(incomingUrl)}
        className={className}
        onClick={handleLinkClick}
        title={title}
        target={targetVal}
        data-locator={dataLocator}
        {...other}
      >
        {children}
      </a>
    ) : (
      <Link href={incomingUrl} as={asLinkPath} shallow={shallow} scroll={scroll}>
        <a
          className={className}
          href={hrefUrl}
          title={title}
          target={targetVal}
          {...(apiConfigObj.isAppChannel
            ? {
                onClick: () => triggerPostMessage(to || asLinkPath),
              }
            : '')}
          data-locator={dataLocator}
          {...other}
        >
          {children || text}
        </a>
      </Link>
    );
  }
  return AnchorComponent;
};

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  asPath: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.string,
  dataLocator: PropTypes.string,
  IsSlugPathAdded: PropTypes.bool,
};

Anchor.defaultProps = {
  asPath: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
  url: '',
  text: '',
  dataLocator: '',
  IsSlugPathAdded: false,
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
