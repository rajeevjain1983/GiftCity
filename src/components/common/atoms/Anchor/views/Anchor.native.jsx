// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import {
  setTestId,
  getLocator,
  configureInternalNavigationFromCMSUrl,
  isWebViewPage,
} from '@tcp/core/src/utils';
import { StyledText } from '../../../../../../styles/globalStyles/StyledText.style';
import {
  UrlHandler,
  navigateToPage,
  validateExternalUrl,
  redirectToInAppView,
} from '../../../../../utils/index.native';
import withStyles from '../../../hoc/withStyles.native';
import { AnchorStyles, AnchorView, AnchorIcon } from '../Anchor.style.native';
import AccessibilityRoles from '../../../../../constants/AccessibilityRoles.constant';

const Icon = require('../../../../../../../mobileapp/src/assets/images/carrot-small-rights.png');

const getOverrideStyle = overrideStyle => {
  return overrideStyle && overrideStyle.color ? { style: overrideStyle } : null;
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is a Anchor component to manage the internal or external .
 */
const Anchor = ({
  anchorVariation,
  text,
  visible,
  children,
  customStyle,
  locator,
  onPress,
  accessibilityLabel,
  justifyContent,
  hitSlop,
  overrideStyle = {},
  ...otherProps
}) => {
  const { url, navigation, onReuqestClose } = otherProps;
  const accessibilityRole = validateExternalUrl(url)
    ? AccessibilityRoles.Link
    : AccessibilityRoles.Button;
  const openUrl = () => {
    if (isWebViewPage(url) && navigation) {
      if (onReuqestClose) {
        onReuqestClose();
      }
      redirectToInAppView(url, navigation);
    } else if (validateExternalUrl(url)) {
      UrlHandler(url);
    } else if (navigation) {
      const cmsValidatedUrl = configureInternalNavigationFromCMSUrl(url);
      navigateToPage(cmsValidatedUrl, navigation);
    }
  };
  const styleProp = getOverrideStyle(overrideStyle);

  if (children) {
    return (
      <TouchableOpacity
        accessibilityRole={accessibilityRole}
        onPress={onPress || openUrl}
        {...otherProps}
        style={customStyle}
        {...setTestId(getLocator(locator))}
      >
        {children}
      </TouchableOpacity>
    );
  }

  if (text) {
    return (
      <AnchorView
        accessibilityRole={accessibilityRole}
        accessible
        onPress={onPress || openUrl}
        accessibilityLabel={accessibilityLabel || text}
        style={customStyle}
        justifyContent={justifyContent}
        {...setTestId(getLocator(locator))}
        hitSlop={hitSlop}
      >
        <StyledText anchorVariation={anchorVariation} {...otherProps} {...styleProp}>
          {text}
        </StyledText>
        {visible && <AnchorIcon source={Icon} />}
      </AnchorView>
    );
  }

  return null;
};

Anchor.propTypes = {
  anchorVariation: PropTypes.string,
  text: PropTypes.string,
  visible: PropTypes.bool,
  children: PropTypes.shape({}),
  customStyle: PropTypes.shape({}),
  locator: PropTypes.string,
  onPress: PropTypes.func,
  accessibilityLabel: PropTypes.string,
  colorName: PropTypes.string,
  margins: PropTypes.string,
  justifyContent: PropTypes.string,
  hitSlop: PropTypes.shape({}),
  overrideStyle: PropTypes.shape({}),
};

Anchor.defaultProps = {
  anchorVariation: '',
  text: '',
  visible: false,
  children: null,
  customStyle: {},
  locator: '',
  onPress: null,
  accessibilityLabel: '',
  colorName: null,
  margins: null,
  justifyContent: '',
  hitSlop: {},
  overrideStyle: {},
};

export default withStyles(Anchor, AnchorStyles);
export { Anchor as AnchorVanilla };
