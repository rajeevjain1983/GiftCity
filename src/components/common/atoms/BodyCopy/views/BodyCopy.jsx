// 9fbef606107a605d69c0edbcd8029e5d
import React from "react";
import { PropTypes } from "prop-types";
import withStyles from "../../../hoc/withStyle";
import styles from "../BodyCopy.style";

const renderComponent = (compProps) => {
  const {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others
  } = compProps;
  return (
    <Component
      className={className}
      data-locator={dataLocator}
      style={customStyle}
      {...others}
    >
      {children}
    </Component>
  );
};

const BodyCopy = (props) => {
  const {
    children,
    className,
    fontFamily,
    component: Component,
    fontSize,
    fontWeight,
    lineHeight,
    letterSpacing,
    textAlign,
    color,
    dataLocator,
    link,
    customStyle,
    ...others
  } = props;

  const compProps = {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others,
  };
  if (!link) {
    return renderComponent(compProps);
  }
};

const fontFamilyPropTypes = PropTypes.oneOf(["primary", "secondary"]);
const fontWeightPropTypes = PropTypes.oneOf([
  "regular",
  "semibold",
  "extrabold",
  "black",
]);
const letterSpacingPropTypes = PropTypes.oneOf([
  "ls271",
  "ls257",
  "ls222",
  "ls167",
  "normal",
]);
const textAlignPropTypes = PropTypes.oneOf(["left", "center", "right"]);
const colorPropTypes = PropTypes.oneOf([
  "text.primary",
  "text.secondary",
  "text.hint",
  "text.disabled",
  "white",
]);
const fontSizesPropTypes = PropTypes.oneOf([
  "fs48",
  "fs42",
  "fs36",
  "fs32",
  "fs28",
  "fs24",
  "fs22",
  "fs18",
  "fs16",
  "fs15",
  "fs14",
  "fs12",
  "fs10",
  "fs9",
]);

BodyCopy.defaultProps = {
  component: "p",
  className: null,
  children: null,
  fontFamily: "primary",
  fontSize: 12,
  fontWeight: "regular",
  lineHeight: "normal",
  letterSpacing: "normal",
  textAlign: "left",
  color: "text.primary",
  link: null,
  customStyle: {},
};

BodyCopy.propTypes = {
  component: PropTypes.elementType,
  className: PropTypes.string,
  children: PropTypes.node,
  fontFamily: PropTypes.oneOfType([
    PropTypes.arrayOf(fontFamilyPropTypes),
    fontFamilyPropTypes,
  ]),
  fontSize: PropTypes.oneOfType([
    PropTypes.arrayOf(fontSizesPropTypes),
    fontSizesPropTypes,
  ]),
  fontWeight: PropTypes.oneOfType([
    PropTypes.arrayOf(fontWeightPropTypes),
    fontWeightPropTypes,
  ]),
  lineHeight: PropTypes.oneOf(["normal"]),
  letterSpacing: PropTypes.oneOfType([
    PropTypes.arrayOf(letterSpacingPropTypes),
    letterSpacingPropTypes,
  ]),
  textAlign: PropTypes.oneOfType([
    PropTypes.arrayOf(textAlignPropTypes),
    textAlignPropTypes,
  ]),
  color: PropTypes.oneOfType([
    PropTypes.arrayOf(colorPropTypes),
    colorPropTypes,
  ]),
  dataLocator: PropTypes.string,
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    target: PropTypes.string,
    text: PropTypes.string,
  }),
  customStyle: PropTypes.shape({}),
};

const StyledBodyCopy = withStyles(BodyCopy, styles);
StyledBodyCopy.defaultProps = BodyCopy.defaultProps;

export { BodyCopy as BodyCopyVanilla };
export default StyledBodyCopy;
