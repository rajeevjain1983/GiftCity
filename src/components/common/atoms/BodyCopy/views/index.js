// 9fbef606107a605d69c0edbcd8029e5d 
export { default } from './BodyCopy';
