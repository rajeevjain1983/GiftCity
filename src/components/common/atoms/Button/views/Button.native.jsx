// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import CustomIcon from '@tcp/core/src/components/common/atoms/Icon';
import { ICON_NAME } from '@tcp/core/src/components/common/atoms/Icon/Icon.constants';
import {
  UrlHandler,
  navigateToPage,
  validateExternalUrl,
  redirectToInAppView,
} from '../../../../../utils/utils.app';
import withStyles from '../../../hoc/withStyles.native';
import {
  style,
  CustomStyleText,
  TouchableOpacityComponent,
  IconContainer,
} from '../Button.style.native';
import {
  getLocator,
  configureInternalNavigationFromCMSUrl,
  isWebViewPage,
} from '../../../../../utils';

const IconComp = values => {
  const {
    showIcon,
    iconName,
    selectedIcon,
    iconColor,
    iconSize,
    selected,
    buttonVariation,
    iconRight,
    customIconStyle,
  } = values;
  if (showIcon) {
    return (
      <IconContainer
        style={customIconStyle}
        buttonVariation={buttonVariation}
        iconRight={iconRight}
      >
        <CustomIcon name={selected ? selectedIcon : iconName} size={iconSize} color={iconColor} />
      </IconContainer>
    );
  }

  return null;
};

const CustomButton = props => {
  const {
    locator,
    text,
    buttonVariation,
    fullWidth,
    customStyle,
    disableButton,
    color,
    fill,
    onPress,
    active,
    selected,
    customTextStyle,
    paddings,
    withNoLineHeight,
    iconFirst,
    fontSize,
    fontWeight,
    fontFamily,
    customBorderStyle,
    useCustomFontStyle,
    ...otherProps
  } = props;
  const textValue = text || '';
  const { url, navigation } = otherProps;
  const openUrl = () => {
    if (isWebViewPage(url) && navigation) {
      redirectToInAppView(url, navigation);
    } else if (validateExternalUrl(url)) {
      UrlHandler(url);
    } else {
      const cmsValidatedUrl = configureInternalNavigationFromCMSUrl(url);
      navigateToPage(cmsValidatedUrl, navigation);
    }
  };
  return (
    <TouchableOpacityComponent
      accessibilityRole="button"
      style={customStyle}
      disabled={disableButton}
      onPress={onPress || openUrl}
      testID={getLocator(locator)}
      {...otherProps}
    >
      {iconFirst ? IconComp(props) : null}
      <CustomStyleText
        fullWidth={fullWidth}
        buttonVariation={buttonVariation}
        color={color}
        fill={fill}
        disableButton={disableButton}
        customBorderStyle={customBorderStyle}
        active={active}
        selected={selected}
        style={customTextStyle}
        paddings={paddings}
        withNoLineHeight={withNoLineHeight}
        fontSize={fontSize}
        fontWeight={fontWeight}
        fontFamily={fontFamily}
        useCustomFontStyle={useCustomFontStyle}
      >
        {textValue}
      </CustomStyleText>
      {!iconFirst ? IconComp(props) : null}
    </TouchableOpacityComponent>
  );
};

CustomButton.propTypes = {
  buttonVariation: PropTypes.string,
  fullWidth: PropTypes.string,
  customStyle: PropTypes.shape({}),
  text: PropTypes.string,
  url: PropTypes.string,
  disableButton: PropTypes.bool,
  locator: PropTypes.string,
  color: PropTypes.string,
  onPress: PropTypes.func,
  fill: PropTypes.string,
  active: PropTypes.bool,
  navigation: PropTypes.shape({}),
  selected: PropTypes.bool,
  theme: PropTypes.shape({}),
  showIcon: PropTypes.bool,
  selectedIcon: PropTypes.string,
  iconName: PropTypes.string,
  iconColor: PropTypes.string,
  iconSize: PropTypes.string,
  withNoLineHeight: PropTypes.bool,
  customTextStyle: PropTypes.shape({}),
  paddings: PropTypes.string,
  iconFirst: PropTypes.bool,
  customIconStyle: PropTypes.shape({}),
  fontSize: PropTypes.string,
  fontWeight: PropTypes.string,
  fontFamily: PropTypes.string,
  customBorderStyle: PropTypes.string,
  useCustomFontStyle: PropTypes.bool,
};

CustomButton.defaultProps = {
  fullWidth: '',
  buttonVariation: 'fixed-width',
  customStyle: {},
  text: '',
  url: '',
  disableButton: false,
  locator: '',
  color: '',
  onPress: null,
  fill: '',
  active: false,
  navigation: {},
  selected: false,
  theme: {},
  iconName: ICON_NAME.chevronDown,
  iconColor: 'gray.800',
  iconSize: 'fs12',
  showIcon: false,
  selectedIcon: ICON_NAME.chevronUp,
  customTextStyle: null,
  paddings: '12px 32px 12px 32px',
  withNoLineHeight: false,
  iconFirst: false,
  customIconStyle: {},
  fontSize: '',
  fontWeight: '',
  fontFamily: '',
  customBorderStyle: '',
  useCustomFontStyle: false,
};

export default withStyles(CustomButton, style);
export { CustomButton as CustomButtonVanilla };
