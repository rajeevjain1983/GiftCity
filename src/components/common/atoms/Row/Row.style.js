// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';

const StyledRow = css`
  ${props =>
    props.theme &&
    props.theme.gridDimensions &&
    props.theme.gridDimensions.gridBreakPointsKeys.map(
      key => `
      ${key !== 'small' ? `@media ${props.theme.mediaQuery[key]} {` : ''}
      ${
        !props.noFlex
          ? `
        display: flex;
        flex-wrap: wrap;
        `
          : ``
      }
      ${
        props.centered
          ? `
        justify-content: center;
        `
          : ``
      }
      ${
        props.fullBleed === true || (props.fullBleed && props.fullBleed[key])
          ? `width: 100%;
            margin-right: 0;
            margin-left: 0;
          `
          : `
          margin-right: ${props.theme.gridDimensions.gridOffsetObj[key]}px;
          margin-left: ${props.theme.gridDimensions.gridOffsetObj[key]}px;
          width: calc(100% - ${props.theme.gridDimensions.gridOffsetObj[key] * 2}px);
          `
      }
      ${key !== 'small' ? `}` : ''}`
    )}

  ${props =>
    props.noLastMargin
      ? ``
      : `
        > *:last-child {
          margin-right: 0;
        }
  `}
  &.content-wrapper {
    margin: 0 auto;
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default StyledRow;
