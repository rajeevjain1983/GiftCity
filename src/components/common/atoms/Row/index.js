// 9fbef606107a605d69c0edbcd8029e5d 
// @flow
export { default } from './views';
