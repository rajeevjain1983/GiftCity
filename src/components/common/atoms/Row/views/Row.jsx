// 9fbef606107a605d69c0edbcd8029e5d
import React from "react";
import { PropTypes } from "prop-types";
import withStyles from "../../../hoc/withStyle";
import StyledRow from "../Row.style";

// An additonal prop 'fullBleed' is added.
// This property ignores the offset of the row and spans across the space of the grid.
const Row = ({
  className,
  children,
  tagName: CustomTag,
  tabIndex,
  customStyle,
  ...otherprops
}) => {
  return (
    <CustomTag
      style={customStyle}
      className={className}
      tabIndex={tabIndex}
      {...otherprops}
    >
      {children}
    </CustomTag>
  );
};

Row.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  tagName: PropTypes.string,
  tabIndex: PropTypes.number,
  otherprops: PropTypes.shape({}),
  customStyle: PropTypes.shape({}),
};

Row.defaultProps = {
  tagName: "div",
  tabIndex: undefined,
  otherprops: {},
  customStyle: {},
};

export default withStyles(Row, StyledRow);
export { Row as RowVanilla };
