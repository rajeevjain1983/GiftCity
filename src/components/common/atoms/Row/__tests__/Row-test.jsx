// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { RowVanilla } from '../views/Row';

describe('Row component', () => {
  it('renders correctly', () => {
    const props = {
      className: 'sample-row',
    };
    const component = shallow(<RowVanilla {...props}>ABCD</RowVanilla>);
    expect(component).toMatchSnapshot();
    expect(component.find('.sample-row')).toHaveLength(1);
  });
});
