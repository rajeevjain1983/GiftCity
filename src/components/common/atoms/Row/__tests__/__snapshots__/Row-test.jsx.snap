// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component renders correctly 1`] = `
<div
  className="sample-row"
  otherprops={Object {}}
  style={Object {}}
>
  ABCD
</div>
`;
